docs/architecture/adr-102-rpc-companion.md

```diff
@@ -0,0 +1,341 @@
+| `net_info`             |                       | Gets network information                  | The request should proxy to the full node since the RPC companion database will not store network information                                                                                                                                   | yes   |
+| `unconfirmed_txs`      | * limit               | Gets the list of unconfirmed transactions | The request should proxy to the full node since the RPC companion database will not store unconfirmed transactions information                                                                                                                  | yes   |
+| `num_unconfirmed_txs`  |                       | Gets data about unconfirmed transactions  | The request should proxy to the full node since the RPC companion database will not store unconfirmed transactions information                                                                                                                  | yes   |
+| `status`               |                       | Gets node status                          | The request should proxy to the full node since the RPC companion database will not store node status information                                                                                                                               | yes   |
+
+> NOTE: The RPC Companion should not implement logic to store data in its database that can modify state in the blockchain such as
+the `broadcast_tx_*` methods.  These requests will proxy to the full node as outlined above.
+
+### High-level architecture
+
+![High-level architecture](images/adr-102-architecture.jpeg)
+
+This diagram shows all the required components for a full RPC Companion solution. The solution implementation contains
+many parts and each one is described below:
+
+### [Ingest Service](#ingest-service)
+
+The **Ingest Service** pulls the data from the full node RPC endpoint and stores the information retrieved in
+the RPC Companion database.
+
+> In the future, if a gRPC interface is implemented in the full node this might be used to pull the data
+from the server.
+
+The **Ingest Service** can control the pruning on the full node via a mechanism to [pruning service](https://github.com/cometbft/cometbft/blob/thane/adr-084-data-companion-pull-api/docs/architecture/adr-101-data-companion-pull-api.md#pruning-service).
+Once the Ingest Service pulls the data from the full node and is able to process it, and it gets an acknowledgement from the database that the data was inserted,
+the **Ingest Service** can communicate with the full node notifying it that a specific height has been processed and set the processed
+height as the `retain height` on the full node signaling this way to the node that this height can be pruned.
```

Comment:
See my previous comment about horizontal scaling

Reply:
Adding to this, by reading further in the text, I understand that the horizontal scaling won't (in principle) affect the Ingest service, which will still be only one instance write to the DB, from which a horizontally-scaled RPC servers will read.

If that is the case, consider explaining this somewhere in the ADR

Reply:
The DB may need to scale horizontally too, but that's something that's taken care of via PostgreSQL configuration.

And the ingest service should not have to scale - there should only ever be a single ingest service for the RPC companion service/cluster.

Reply:
Yes, Thane is right. One Ingest / Multiple RPC if needed. I'll try to emphasize that the ingest service should be a "singleton"